packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -681,10 +691,17 @@ export class ModularChangeFamily
 			? [{ revision: revisionForInvert, rollbackOf: change.revision }]
 			: [{ revision: revisionForInvert }];
 
+		// If this invert is the result of a revert, then we need to ensure noChangeConstraintOnRevert
+		// is applied under noChangeConstraint, so that if this is then rebased, the noChangeConstraint will be violated.
+		const noChangeConstraint = isRevert ? change.change.noChangeConstraintOnRevert : undefined;
```

Comment:
There's no need to worry about whether or not this is a revert because when it's not a revert (meaning it is a rollback) the changeset will never be rebased and the constraint will stay valid.